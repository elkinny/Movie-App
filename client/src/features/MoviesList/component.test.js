import MoviesListComponent from './component';
import React from 'react';
import { shallow } from 'enzyme';

describe('MoviesListComponent: ', () => {
  it('1. To match snapshot - no movies', () => {
    const props = {
      movies: [],
    };
    const component = shallow(<MoviesListComponent {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('2. To match snapshot - with movies', () => {
    const props = {
      movies: [
        {
          id: 1,
          runtime: '120min',
          vote_average: '4.0',
          title: 'Captain Marvel',
          genres: ['Action'],
          release_date: '2019',
          overview:
            'Carol Danvers becomes one of the universe`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races.',
          link: 'https://www.youtube.com/watch?v=0LHxvxdRnYc',
          poster_path:
            'https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SY1000_CR0,0,675,1000_AL_.jpg',
        },
        {
          id: 2,
          runtime: 120,
          vote_average: 5,
          title: 'Captain Marvel',
          genres: ['Action'],
          release_date: '2019',
          overview:
            'Carol Danvers becomes one of the universe`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races.',
          link: 'https://www.youtube.com/watch?v=0LHxvxdRnYc',
          poster_path:
            'https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SY1000_CR0,0,675,1000_AL_.jpg',
        },
      ],
    };
    const component = shallow(<MoviesListComponent {...props} />);
    expect(component).toMatchSnapshot();
  });
});
