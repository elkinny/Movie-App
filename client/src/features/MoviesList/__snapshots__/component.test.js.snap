// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesListComponent:  1. To match snapshot - no movies 1`] = `
<div
  className="movie-list"
>
  <span
    className="movie-list__message"
  >
     No movies that match your query 
  </span>
</div>
`;

exports[`MoviesListComponent:  2. To match snapshot - with movies 1`] = `
<div
  className="movie-list"
>
  <MoviesListItemComponent
    genres={
      Array [
        "Action",
      ]
    }
    id={1}
    key="Captain Marvel_0"
    link="https://www.youtube.com/watch?v=0LHxvxdRnYc"
    overview="Carol Danvers becomes one of the universe\`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races."
    poster_path="https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SY1000_CR0,0,675,1000_AL_.jpg"
    release_date="2019"
    runtime="120min"
    title="Captain Marvel"
    vote_average="4.0"
  />
  <MoviesListItemComponent
    genres={
      Array [
        "Action",
      ]
    }
    id={2}
    key="Captain Marvel_1"
    link="https://www.youtube.com/watch?v=0LHxvxdRnYc"
    overview="Carol Danvers becomes one of the universe\`s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races."
    poster_path="https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SY1000_CR0,0,675,1000_AL_.jpg"
    release_date="2019"
    runtime={120}
    title="Captain Marvel"
    vote_average={5}
  />
</div>
`;
